rbd/snapshot_nautilus.go

```diff
@@ -66,3 +66,37 @@ func (image *Image) GetParentInfo(p_pool, p_name, p_snapname []byte) error {
 
 	return nil
 }
+
+// ssize_t rbd_list_children(rbd_image_t image, char *pools, size_t *pools_len,
+//               char *images, size_t *images_len);
+func (image *Image) ListChildren() (pools []string, images []string, err error) {
+	if err := image.validate(imageIsOpen); err != nil {
+		return nil, nil, err
+	}
+
+	size := C.size_t(0)
+	ret := C.rbd_list_children3(image.image, nil, &size)
+	if ret < 0 && ret != -C.ERANGE {
+		return nil, nil, RBDError(ret)
+	} else if ret > 0 {
+		return nil, nil, fmt.Errorf("rbd_list_children3() returned %d, expected 0", ret)
+	} else if ret == 0 && size == 0 {
+		return make([]string, 0), make([]string, 0), nil
```

Comment:
`return nil, nil, nil` would be shorter, more succinct, and equivalent

Reply:
Done